codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -403,4 +403,14 @@
      * Indicates that a {@code GOAWAY} was sent to the remote endpoint and sets the last known stream.
      */
     void goAwaySent(int lastKnownStream, long errorCode, ByteBuf message);
+
+    /**
+     * Indicates that headers has been sent to the remote on this stream.
+     */
+    void headersSent(int streamId, boolean sent);
```

Comment:
we can just move this method onto the `Http2Stream` interface to avoid duplicate map lookups, and this way we can also avoid using the stream's property map.